src/test/java/it/eng/idsa/businesslogic/service/impl/DapsTokenProviderServiceImplTest.java

```diff
@@ -0,0 +1,105 @@
+package it.eng.idsa.businesslogic.service.impl;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.text.ParseException;
+import java.util.Date;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import io.jsonwebtoken.JwtBuilder;
+import io.jsonwebtoken.Jwts;
+import it.eng.idsa.businesslogic.service.DapsService;
+
+public class DapsTokenProviderServiceImplTest {
+
+	@Mock
+	private DapsService dapsService;
+
+	@InjectMocks
+	private DapsTokenProviderServiceImpl dapsTokenProviderServiceImpl;
+
+	private String dapsToken;
+
+	@BeforeEach
+	public void setup() throws ParseException {
+		MockitoAnnotations.initMocks(this);
+		ReflectionTestUtils.setField(dapsTokenProviderServiceImpl, "tokenCaching", true);
+		dapsToken = generateToken(1000*60*60);
+		when(dapsService.getJwtToken()).thenReturn(dapsToken);
+
+	}
+
+	@Test
+	public void tokenNullTest() {
+		assertNotNull(dapsTokenProviderServiceImpl.provideToken());
+		verify(dapsService).getJwtToken();
+
+	}
+
+	@Test
+	public void tokenExpiredAndFetchedNewTokenTest() {
```

Comment:
Maybe rename test to fetchNewTokenWhenExpiredTest?

Reply:
Sure, no problem.